src/parser/parser.yy

```diff
@@ -2609,6 +2606,45 @@ comment_prop
     }
     ;
 
+opt_index_param_list
+    : %empty {
+        $$ = nullptr;
+    }
+    | index_param_list {
+        $$ = $1;
+    }
+    ;
+
+index_param_list
+    : index_param_item {
+        $$ = new IndexParamList();
+        $$->add($1);
+    }
+    | index_param_list COMMA index_param_item {
+        $$ = $1;
+        $$->add($3);
+    }
+    ;
+
+index_param_item
+    : comment_prop_assignment {
```

Comment:
`comment_prop` is better than `comment_prop_assignment`, refer to https://dev.mysql.com/doc/refman/8.0/en/create-index.html

Reply:
ok, good!

Reply:
@Shylock-Hg  WDYT ?

Reply:
It'ok. But better modify it in independent PR.